/**********************************************************************************************************************
【文件名】                                  msgdetaildlg.cpp
【功能模块和目的】                            MsgDetailDlg类中成员函数的实现, 包括槽函数、接口函数,主要展现消息内容
【开发者及日期】                             门恺文2019.7.27
【更改记录】
**********************************************************************************************************************/

#include "msgdetaildlg.h"
#include "ui_msgdetaildlg.h"

/***********************************************************************************************************************
【函数名称】                                 MsgDetailDlg()
【函数功能】                                 类MsgDetailDlg的构造函数，在创建对象时被调用
【参数】                                    QWidget *parent，输入参数
【返回值】                                   无
【开发者及日期】                              门恺文 2019.7.27
【更改记录】
***********************************************************************************************************************/

MsgDetailDlg::MsgDetailDlg(QWidget *parent) :
    QDialog(parent),
    ui(new Ui::MsgDetailDlg)
{
    ui->setupUi(this);
    ui->plainTextEditContent->setReadOnly(true);
}

/***********************************************************************************************************************
【函数名称】                                 ~MsgDetailDlg()
【函数功能】                                 类MsgDetailDlg的析构函数，在对象被销毁时调用
【参数】                                    ui
【返回值】                                   无
【开发者及日期】                              门恺文2019.7.27
【更改记录】
***********************************************************************************************************************/

MsgDetailDlg::~MsgDetailDlg()
{
    delete ui;
}

/***********************************************************************************************************************
【函数名称】                                 setMsgId()
【函数功能】                                 成员msgId的设置函数，改变页面，使页面对应特定消息
【参数】                                     value, int, 输入参数
【返回值】                                   无
【开发者及日期】                              门恺文2019.7.27
【更改记录】
***********************************************************************************************************************/

void MsgDetailDlg::setMsgId(int value)
{
    msgId = value;
    //获取消息编号后设置消息详情
    ui->labFront->setText(tr("消息") + QString::number(msgId));
    ui->labThemeDetail->setText(msgList[msgId].sTheme());
    ui->plainTextEditContent->setPlainText(msgList[msgId].sMessage());
}

/***********************************************************************************************************************
【函数名称】                                 setSenderName()
【函数功能】                                 成员senderName的设置函数，显示发送者
【参数】                                     value, int, 输入参数
【返回值】                                   无
【开发者及日期】                              门恺文2019.7.27
【更改记录】
***********************************************************************************************************************/

void MsgDetailDlg::setSenderName(const QString &value)
{
    senderName = value;
    //获取发送者用户名后设置界面上发送者用户名
    ui->labSenderDetail->setText(senderName);
}
